### PR TITLE
Add splashColor property for ListTile widget

### LINKED ISSUE #99898: [Proposal] Allow customization of ListTile Splash
## Use case
When pressing down on a ListTile, it obtains a grey background. I'd like to be able to specify which color this background should be but none of the `color` properties of `ListTile` seem to change that particular color.

## Proposal
Add a new color property to `ListTile` (and related widgets like `ListTileThemeData` , maybe `InkWell`).

### PR DESCRIPTION
**Pull request by M97Chahboun**
Adds a `splashColor` property to `ListTile` widget & test it,

Fixes #99898

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.
**I can't pass all existing tests because my Mac slow, but I passed ListTile test.**

### CODE REVIEW

**Comment by M97Chahboun** [comment]
@HansMuller 
Check this PR & Thank you.

File: packages/flutter/test/material/list_tile_test.dart [hunk 1]
```diff
@@ -1294,6 +1294,33 @@ void main() {
     );
   });
 
+  testWidgets('ListTile can be splashed and has correct splash color', (WidgetTester tester) async {
+    final Widget buildApp = MaterialApp(
+        home: Material(
+          child: Center(
+            child: StatefulBuilder(builder: (BuildContext context, StateSetter setState) {
```
Comment (HansMuller) [inline packages/flutter/test/material/list_tile_test.dart:1301]: This StatefulBuilder isn't needed.

File: packages/flutter/test/material/list_tile_test.dart [hunk 2]
```diff
@@ -1294,6 +1294,33 @@ void main() {
     );
   });
 
+  testWidgets('ListTile can be splashed and has correct splash color', (WidgetTester tester) async {
+    final Widget buildApp = MaterialApp(
+        home: Material(
```
Comment (HansMuller) [inline packages/flutter/test/material/list_tile_test.dart:1299]: Two-space indent

File: packages/flutter/test/material/list_tile_test.dart [hunk 3]
```diff
@@ -1294,6 +1294,33 @@ void main() {
...
+          child: Center(
+            child: StatefulBuilder(builder: (BuildContext context, StateSetter setState) {
+              return Container(
```
Comment (HansMuller) [inline packages/flutter/test/material/list_tile_test.dart:1302]: This container doesn't appear to be needed either.

**Review by HansMuller — COMMENTED** [review]
This looks pretty close, just needs some small tweaks.

**Comment by M97Chahboun** [comment]
Thank you for your help 👍🏾👏🏾

**Comment by guidezpl** [comment]
Another merge should do the trick, also it looks like you're missing the CLA.

**Comment by HansMuller** [comment]
This PR looks like a reasonable and narrowly focused change however it has failed many presubmit tests. It may be possible to get it going by rebasing with the latest code.

**Comment by M97Chahboun** [comment]
OK Thank you I will try to rebase it

**Review by HansMuller — APPROVED** [review]
LGTM

**Review by guidezpl — APPROVED** [review]

**Pull request merged**